test/leaflike/functional/bookmarks.clj

```diff
@@ -37,7 +37,7 @@
       (is (= (:error-msg flash) "Invalid bookmark"))))
 
   (testing "list all bookmarks"
-    (let [response (bm/list-all {:session {:username (:username user)}})]
+    (let [response (bm/fetch-bookmarks (:username user) {:page 1})]
```

Comment:
@samrat Why is `list-all` getting deprecated in this test? A functional test should not have to be concerned with `{:page 1}` to test for all bookmarks

Reply:
@trycatcher I've replaced `list-all` with a more general function `fetch-bookmarks` which does pagination. This pagination-enabled handler is all that's accessible to the end user(there's no longer a page for the user to view all of his/her bookmarks in one page). So I think it makes sense to change this test to use the same function as well.

Reply:
@samrat The calling code (test here) should have as less knowledge of the called code as possible. A view needs to be aware of pagination, but a caller that wants to get all bookmarks doesn't need that.  This is leaking details. You can have a a wrapper around `fetch-bookmarks` that is called by the test

Reply:
Fixed this